design/accepted/gc_plus/gc-plus-cloud.md

```diff
@@ -0,0 +1,64 @@
+# Managed Uncommitted Garbage Collection for lakeFS cloud - Execution Plan
+
+## Goals
+1. Run UGC for lakeFS cloud users
+2. Allow Cloud users to clean up storage space by deleting objects from the underlying storage for cost efficiency
+3. Support new and existing cloud users (both repository structures)
+
+## Non-Goals
+1. Remove lakeFS limitations during the UGC job run
+2. Improve UGC OSS support
+
+### User Story
+For every cloud installation, The UGC will run as default as a background job (sweep=true) without visibility to the user to control its configuration.
+We should have to ability to control the MUGC for the user - change configuration, or stop it.
+
+### Permissions
+Create default UGC user with permissions for every new cloud env (The permissions to Enigma was added to the GC user)
+- lakeFS permissions - prepare uncommitted gc
+- Bucket permissions - role to list, write, delete (can get the same role as the GC)
```

Comment:
As opposed to UGC configurations that are not visible to the user, this part is visible as we need them to create roles. Product-wise, I think that it will make more sense to tie the required permissions to the lakefs cloud role. 

What I'm suggesting is not to rely on the user to define the role for CGC but make them add the permissions to the role they defined anyways and holds for any resource managed by lakeFS.

Reply:
If we go that path we need to clarify that defining a gc role is a mandatory step of env setup, which is currently not a must-do step